### PR TITLE
[MIG] Migrated 'base_gengo' module into new api.

### PR DESCRIPTION
**Pull request by psa-odoo**
Task: https://www.odoo.com/web#id=12209&view_type=form&model=project.task&menu_id=3942&action=327
Pad: https://pad.odoo.com/p/r.4YJMn6R8itJDcfnQ

### CODE REVIEW

File: addons/base_gengo/controller/gengo_callback.py [hunk 1]
```diff
@@ -1,33 +1,29 @@
...
+        IrConfigParamSudo = request.env['ir.config_parameter'].sudo()
+        return IrConfigParamSudo.get_param(request.env['base.gengo.translations'].GENGO_KEY, default="")
 
     @http.route('/website/gengo_callback', type='http', auth='none')
     def gengo_callback(self, **post):
         print "IN website/gengo_callback"
-        cr, uid, context = request.cr, openerp.SUPERUSER_ID, request.context
-        translation_pool = request.registry['ir.translation']
+        IrTranslationSudo = request.env['ir.translation'].sudo()
         if post and post.get('job') and post.get('pgk'):
```
Comment (ara-odoo) [inline addons/base_gengo/controller/gengo_callback.py]: No need to check if post.

File: addons/base_gengo/models/ir_translation.py [hunk 2]
```diff
@@ -58,21 +38,23 @@
...
 
 
-class ir_translation(osv.Model):
+class IrTranslation(models.Model):
     _name = "ir.translation"
     _inherit = "ir.translation"
 
-    _columns = {
-        'gengo_comment': fields.text("Comments & Activity Linked to Gengo"),
-        'order_id': fields.char('Gengo Order ID', size=32),
```
Comment (ara-odoo) [inline addons/base_gengo/models/ir_translation.py]: Remove size attribute.

File: addons/base_gengo/models/ir_translation.py [hunk 3]
```diff
@@ -58,21 +38,23 @@
...
-        "gengo_translation": fields.selection([('machine', 'Translation By Machine'),
-                                             ('standard', 'Standard'),
-                                             ('pro', 'Pro'),
-                                             ('ultra', 'Ultra')], "Gengo Translation Service Level", help='You can select here the service level you want for an automatic translation using Gengo.'),
-    }
+    gengo_comment = fields.Text(string="Comments & Activity Linked to Gengo")
+    order_id = fields.Char(string='Gengo Order ID', size=32)
```
Comment (ara-odoo) [inline addons/base_gengo/models/ir_translation.py]: Remove size attribute.

File: addons/base_gengo/controller/gengo_callback.py [hunk 4]
```diff
@@ -1,33 +1,29 @@
...
+            if post.get('pgk') != self.get_gengo_key():
                 return Response("Bad authentication", status=104)
             job = json.loads(post['job'], 'utf-8')
             tid = job.get('custom_data', False)
             if (job.get('status') == 'approved') and tid:
-                term = translation_pool.browse(cr, uid, int(tid), context=context)
+                term = IrTranslationSudo.browse(int(tid))
                 if term.src != job.get('body_src'):
                     return Response("Text Altered - Not saved", status=418)
                 domain = [
```
Comment (ara-odoo) [inline addons/base_gengo/controller/gengo_callback.py]: Indent the domain to be more readable.

File: addons/base_gengo/models/res_company.py [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+
+from openerp import fields, models
+
+
+class ResCompany(models.Model):
+    _name = "res.company"
+    _inherit = "res.company"
+
+    gengo_private_key = fields.Text(string="Gengo Private Key",
```
Comment (ara-odoo) [inline addons/base_gengo/models/res_company.py]: Remove the string parameter here, as the ORM will generate a capitalized version of the field name by default

File: addons/base_gengo/wizard/base_gengo_translations.py [hunk 6]
```diff
@@ -39,33 +18,31 @@
...
-        'sync_limit': 20
-    }
+
+    sync_type = fields.Selection([
+        ('send', 'Send New Terms'),
+        ('receive', 'Receive Translation'),
+        ('both', 'Both')], string="Sync Type", required=True, default='both')
+    lang_id = fields.Many2one('res.lang', string='Language', required=True)
+    sync_limit = fields.Integer(string="No. of terms to sync", default=20)
+
```
Comment (ara-odoo) [inline addons/base_gengo/wizard/base_gengo_translations.py]: pep-8 too many blank lines.

File: addons/base_gengo/wizard/base_gengo_translations.py [hunk 7]
```diff
@@ -172,42 +146,37 @@ def _update_terms_job(self, cr, uid, job, context=None):
...
+    def pack_jobs_request(self, terms):
         ''' prepare the terms that will be requested to gengo and returns them in a dictionary with following format
             {'jobs': {
                 'term1.id': {...}
                 'term2.id': {...}
                 }
             }'''
-        base_url = self.pool.get('ir.config_parameter').get_param(cr, uid, 'web.base.url')
-        translation_pool = self.pool.get('ir.translation')
+        BaseUrl = self.env['ir.config_parameter'].get_param('web.base.url')
```
Comment (rgo-odoo) [inline addons/base_gengo/wizard/base_gengo_translations.py]: should be 'base_url' as it does not hold model registry and same remark for 'User' below.

File: addons/base_gengo/wizard/base_gengo_translations.py [hunk 8]
```diff
@@ -76,14 +51,14 @@ def gengo_authentication(self, cr, uid, context=None):
             by the cron) or in a dialog box (if requested by the user), thus it's important to return it
             translated.
         '''
-        user = self.pool.get('res.users').browse(cr, 1, uid, context=context)
-        if not user.company_id.gengo_public_key or not user.company_id.gengo_private_key:
+        User = self.env.user
```
Comment (rgo-odoo) [inline addons/base_gengo/wizard/base_gengo_translations.py]: User ==> user

File: addons/base_gengo/wizard/base_gengo_translations.py [hunk 9]
```diff
@@ -91,57 +66,56 @@ def gengo_authentication(self, cr, uid, context=None):
...
-        translation_pool = self.pool.get('ir.translation')
-        flag, gengo = self.gengo_authentication(cr, uid, context=context)
+        IrTranslation = self.env['ir.translation']
+        flag, gengo = self.gengo_authentication()
         if not flag:
             _logger.warning("%s", gengo)
         else:
             offset = 0
-            all_translation_ids = translation_pool.search(cr, uid, [('state', '=', 'inprogress'), ('gengo_translation', 'in', ('machine', 'standard', 'pro', 'ultra')), ('order_id', "!=", False)], context=context)
+            Translations = IrTranslation.search([
```
Comment (rgo-odoo) [inline addons/base_gengo/wizard/base_gengo_translations.py]: recordset should be stored in lowercase variables.

**Comment by rgo-odoo** [comment]
Need to rename:

controller/base_gengo.py ==> controller/main.py
data/gengo_sync_schedular_data.xml ==> data/ir_cron_data.xml
views/ir_translation.xml ==> views/ir_translation_views.xml
views/res_company_view.xml ==> views/res_company_views.xml

**Comment by psa-odoo** [comment]
Done

File: addons/base_gengo/controller/main.py [hunk 10]
```diff
@@ -1,33 +1,29 @@
...
-    def get_gengo_key(self, cr):
-        icp = request.registry['ir.config_parameter']
-        return icp.get_param(cr, SUPERUSER_ID, request.registry['base.gengo.translations'].GENGO_KEY, default="")
+    def get_gengo_key(self):
+        IrConfigParamSudo = request.env['ir.config_parameter'].sudo()
+        return IrConfigParamSudo.get_param(request.env['base.gengo.translations'].GENGO_KEY, default="")
 
     @http.route('/website/gengo_callback', type='http', auth='none')
     def gengo_callback(self, **post):
         print "IN website/gengo_callback"
```
Comment (hmo-odoo) [inline addons/base_gengo/controller/main.py]: remove print statement

**Pull request closed without merging**